src/rpc/net.cpp

```diff
@@ -875,8 +879,13 @@ static RPCHelpMan getnodeaddresses()
     const int count{request.params[0].isNull() ? 1 : request.params[0].get_int()};
     if (count < 0) throw JSONRPCError(RPC_INVALID_PARAMETER, "Address count out of range");
 
+    const Network network{request.params[1].isNull() ? Network::NET_MAX : ParseNetwork(request.params[1].get_str())};
+    if (network == NET_UNROUTABLE) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid network (not recognized)");
```

Comment:
Maybe print the unrecognized string:
```suggestion
        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Network not recognized: %s", request.params[1].get_str());
```

Reply:
Good idea! Done.